plugins/by-name/treesitter-textobjects/deprecations.nix

```diff
@@ -0,0 +1,105 @@
+lib: {
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "treesitter-textobjects"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+
+      processKeymaps = lib.mapAttrs (
+        key: mapping:
+        if lib.isString mapping then
+          mapping
+        else
+          {
+            inherit (mapping) query;
+            query_group = mapping.queryGroup;
+            inherit (mapping) desc;
+          }
+      );
+      mkChangedKeymapOption =
+        oldSubPath: newSubPath:
+        let
+          oldPath = basePluginPath ++ oldSubPath;
+          newPath = settingsPath ++ newSubPath;
+        in
+        lib.mkChangedOptionModule oldPath newPath (
+          config:
+          let
+            oldValue = lib.getAttrFromPath oldPath config;
+          in
+          (
+            if oldValue == null then
+              lib.id
+            else
+              lib.warn ''
+                WARNING: `${lib.showOption newPath}` will not convert the sub-option names from camelCase to snake_case.
```

Comment:
Not sure I understand the warning here. Aren't the new options being converted manually to snake_case in what you pass in below?

Reply:
Not the sub-options. Each keymap value can be either a string or an attrs.
In the latter, we used to map `queryGroup` to `query_group`. Users should not blindly copy-pase
```nix
select.keymaps.foo = {
  query  = "X";
  queryGroup = "Y";
  desc = "Z";
};
```
to
```nix
settings.select.keymaps.foo = {
  query  = "X";
  queryGroup = "Y";
  desc = "Z";
};
```
but to
```nix
settings.select.keymaps.foo = {
  query  = "X";
  query_group = "Y";
  desc = "Z";
};
```